kore/src/Kore/ASTVerifier/SentenceVerifier.hs

```diff
@@ -357,12 +387,141 @@ verifyAxiomSentence sentence =
                 (freeVariables sentence)
                 (sentenceAxiomAttributes sentence)
         State.modify $ addAxiom verified attrs
+        case fromSentenceAxiom (attrs, verified) of
```

Comment:
This block of code has detailed knowledge of `Equation`. Let's move it to `Kore.Equation.Validate`, and make sure it gets called before `addAxiom` (above).